src/include/stir/recon_buildblock/PLSPrior.h

```diff
@@ -0,0 +1,213 @@
+
+  //! set/get anatomical pointers
+  void set_anatomical_image_sptr(const shared_ptr<DiscretisedDensity<3,elemT> >&);
+  shared_ptr<DiscretisedDensity<3,elemT> > get_anatomical_image_sptr() const;
+
+protected:
+  //! can be set during parsing to restrict the gradient calculation to the 2D case
+  bool only_2D;
+  //! filename prefix for outputing the gradient whenever compute_gradient() is called.
+  /*! An internal counter is used to keep track of the number of times the
+     gradient is computed. The filename will be constructed by concatenating
+     gradient_filename_prefix and the counter.
+  */
+  std::string gradient_filename_prefix;
+
+  //! Filename for the \f$\kappa\f$ image that will be read by post_processing()
+  std::string kappa_filename;
+  std::string anatomical_filename;
+
+  double eta, alpha;
+
+  virtual void set_defaults();
+  virtual void initialise_keymap();
+  virtual bool post_processing();
+ private:
+
+   virtual Succeeded set_up();
+
+  //! compute the component x, y or z of the image gradient using forward difference
+  static void compute_image_gradient_element(DiscretisedDensity<3,elemT> & image_gradient_elem,
+                                      int direction,
+                                      const DiscretisedDensity<3,elemT> & image );
+
+  //! compute normalisation for the gradient of the anatomical image (Eq. (5) of the paper)
+  void compute_normalisation_anatomical_gradient(DiscretisedDensity<3, elemT> &norm_im_grad,
+                                            const DiscretisedDensity<3,elemT> &image_grad_z,
+                                            const DiscretisedDensity<3,elemT> &image_grad_y,
+                                            const DiscretisedDensity<3,elemT> &image_grad_x);
+  //! Inner product in Eq. (9) of the paper but also the penalty function.
+  void compute_inner_product_and_penalty(DiscretisedDensity<3,elemT> &inner_product,
+                                           DiscretisedDensity<3,elemT> &penalty,
+                                         DiscretisedDensity<3,elemT> &pet_im_grad_z,
+                                         DiscretisedDensity<3,elemT> &pet_im_grad_y,
+                                         DiscretisedDensity<3,elemT> &pet_im_grad_x,
+                        const DiscretisedDensity<3,elemT> &pet_image);
+
+  shared_ptr<DiscretisedDensity<3,elemT> > anatomical_grad_x_sptr;
+  shared_ptr<DiscretisedDensity<3,elemT> > anatomical_grad_y_sptr;
+  shared_ptr<DiscretisedDensity<3,elemT> > anatomical_grad_z_sptr;
+  shared_ptr<DiscretisedDensity<3,elemT> > anatomical_sptr;
+  shared_ptr<DiscretisedDensity<3,elemT> > norm_sptr;
+  shared_ptr<DiscretisedDensity<3,elemT> > kappa_ptr;
+  void set_anatomical_grad_sptr(const shared_ptr<DiscretisedDensity<3,elemT> >&, int);
+  void set_anatomical_grad_norm_sptr(const shared_ptr<DiscretisedDensity<3,elemT> >&);
+  void set_eta(const double&);
```

Comment:
can you make `set_eta` and `set_alpha` public? Will be useful for Python

Reply:
Sorry I clearly committed the wrong files.
Sorry about that.